### PR TITLE
Cap aggregation row count at input row count

### PR DESCRIPTION
**Pull request by rschlussel-zz**
The row count for aggregations is estimated as the product of the ndvs of each
grouping column. However, the number of output rows can't be greater
than the number of input rows.  When the product of the number of ndvs
is greater than the the input row count, estimate the output row count
as equal to the input row count.

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/cost/PlanNodeStatsEstimateMath.java [hunk 1]
```diff
@@ -148,7 +148,7 @@ public static PlanNodeStatsEstimate intersect(PlanNodeStatsEstimate left, PlanNo
                             .setNullsFraction(min(leftSymbolStats.getNullsFraction(), rightSymbolStats.getNullsFraction()))
                             .build());
         }
-
+        statsBuilder.setOutputRowCount(left.getOutputRowCount() + right.getOutputRowCount());  // this is the maximum row count;
```
Comment (losipiuk) [inline presto-main/src/main/java/com/facebook/presto/cost/PlanNodeStatsEstimateMath.java:151]: this seems unrelated

**Review by losipiuk — COMMENTED** [review]
looks good modulo one quesion

File: presto-main/src/main/java/com/facebook/presto/cost/PlanNodeStatsEstimateMath.java (hunk 1 above)
Comment (rschlussel-zz) [reply presto-main/src/main/java/com/facebook/presto/cost/PlanNodeStatsEstimateMath.java:151]: it's needed because otherwise the input to the group by doesn't have a row count (and then the output size is estimated at NaN)

Comment (losipiuk) [reply presto-main/src/main/java/com/facebook/presto/cost/PlanNodeStatsEstimateMath.java:151]: I mean that it seems like this should be separate commit. As it just fixes the issue of missing output row count estimate for INTERSECT (btw - test would be nice for that). And the commit in this PR says about adding capping of output rowCount to input rowCount.

Comment (rschlussel-zz) [reply presto-main/src/main/java/com/facebook/presto/cost/PlanNodeStatsEstimateMath.java:151]: It was caught by unit tests for TestIntersectStatsRule: https://travis-ci.org/Teradata/presto/jobs/259865594 .

Comment (losipiuk) [reply presto-main/src/main/java/com/facebook/presto/cost/PlanNodeStatsEstimateMath.java:151]: Oh - and previously groupBy set the outputRowCount unconditionally. Ok. I get it now. Looks good. Thanks.

**Review by losipiuk — APPROVED** [review]

File: presto-main/src/main/java/com/facebook/presto/cost/PlanNodeStatsEstimateMath.java (hunk 1 above)
Comment (rschlussel-zz) [reply presto-main/src/main/java/com/facebook/presto/cost/PlanNodeStatsEstimateMath.java:151]: Since these are intermediate stats internal to this computation, it only became a problem when we started using the rowCount in groupBy.  So, I can keep it as a separate commit, but IMO it's directly related to the rest of the changes (e.g. it can't be tested independently).

Comment (losipiuk) [reply presto-main/src/main/java/com/facebook/presto/cost/PlanNodeStatsEstimateMath.java:151]: Yeah. I agree. Keep it this way.

**Comment by rschlussel-zz** [comment]
I guess I didn't refresh github :).  Thanks for the review!  I'll merge when travis passes.
